pkg/apis/network/v1/egressrouter_types.go

```diff
@@ -0,0 +1,124 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EgressRouter is a feature allowing the user to define an egress router
+// that acts as a bridge between pods and external systems. The egress router runs
+// a service that redirects egress traffic originating from a pod or a group of
+// pods to a remote external system or multiple destinations as per configuration.
+
+//  More specifically, given an EgressRouter with <name>, the CNO will create and manage:
+// - A service called <name>
+// - An egress pod called <name>
+// - An NAD called <name>
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EgressRouter is a single egressrouter pod configuration object.
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=egressrouters,scope=Namespaced
+type EgressRouter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	Spec EgressRouterSpec `json:"spec"`
+
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status []EgressRouterStatus `json:"status,omitempty"`
+}
+
+// EgressRouterSpec contains the configuration for an egress router.
+// +k8s:openapi-gen=true
+// +kubebuilder:validation:Required
+type EgressRouterSpec struct {
+	// The name of the network attachment definition
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
```

Comment:
Do we need this? Or can we just use the name of the CR itself?

Reply:
I think we can use the name of the CR itself, name of CR == name of NAD == name of router == name of pod == name of svc. @danielmellado : I think we talked about this before and you were not particular on this right ? So I guess we can safely remove this field and stick to using the name of the CRD?

Reply:
works for me in that part, I mean, hopefully, both, NAD, svc and pod would be hidden from the user anyway.

Reply:
cool, ok so I'm removing `spec.name`.
Consensus: to use the CR's name.